plasmapy/simulation/particle_tracker.py

```diff
@@ -455,16 +424,47 @@ def __init__(
         # Update the `tracker` attribute so that the stop condition & save routine can be used
         termination_condition.tracker = self
 
-        if save_routine is not None:
-            save_routine.tracker = self
+        save_routine.tracker = self
 
         self.termination_condition = termination_condition
         self.save_routine = save_routine
 
-    @property
-    def num_grids(self):
-        """The number of grids specified at instantiation."""
-        return len(self.grids)
+    def _set_time_step_attributes(self, dt, termination_condition, save_routine):
+        """Determines whether the simulation will follow a synchronized or adaptive time step."""
+
+        if not isinstance(termination_condition, AbstractTerminationCondition):
```

Comment:
These are input validation - I think they should go in the constructor or in another validator, not in this `set` method